### DISCUSSION TITLE
Pin a Cask?

### DISCUSSION THREAD

**Discussion by bstopp** [Homebrew/discussions#97] [Everyday usage]
Hell all - I'm running into an issue with Virtual box and have downgraded my cask to the version I want.  But how do I _pin_ the cask?

I've tried every permutation of `brew pin` but nothing seems to work.

Any help would be appreciated. Thanks!

**Comment by crunchtime-ali** [comment]
@bstopp Homebrew itself does not provide means to pin a cask but there is a way to accomplish this. The tool [brew-cask-upgrade](https://github.com/buo/homebrew-cask-upgrade) which can be installed via simply tapping (`brew tap buo/cask-upgrade`) can pin casks properly if you upgrade your casks with it (which I highly recommend because it is a much nicer experience).

[![asciicast](https://asciinema.org/a/DlXUmiFFVnDhIDe2tCGo3ecLW.png)](https://asciinema.org/a/DlXUmiFFVnDhIDe2tCGo3ecLW)

**Comment by Kolvir73** [comment]
I like this solution, but how do I upgrade only formula then with 'brew upgrade' and not casks? I don't want to have to upgrade all outdated formula individually just to be able to upgrade casks separately with brew-cask-upgrade.

Would 'brew upgrade --formula' ignore the casks?

**Reply by gromgit** [reply]
```sh
$ brew upgrade --help
Usage: brew upgrade [options] [formula|cask]
...
      --formula, --formulae        Treat all named arguments as formulae. If
                                   no named arguments are specified, upgrade
                                   only outdated formulae.
```

**Comment by Kolvir73** [comment]
I saw that but it wasn't exactly clear that it ignored casks in that case.

**Reply by gromgit** [reply]
If you can think of a clearer and non-redundant way to phrase it, I'm sure the Homebrew maintainers would welcome a PR. In my mind:
* `--help` text should be as succinct as possible
* "only X" implies "nothing else but X".

**Comment by tddschn** [comment]
Maybe add `auto_updates true` stanza to the cask file you wish to pin?

**Reply by gromgit** [reply]
That may cause merge conflicts at a later date, so not recommended.
